lib/query.js

```diff
@@ -4007,12 +4013,83 @@ Query.prototype._findAndModify = function(type, callback) {
  */
 
 function _completeOneLean(doc, res, opts, callback) {
+  if (opts.lean && opts.lean.transform) {
+    for (const key of Object.keys(doc)) {
+      if (Array.isArray(doc[key])) {
+        doc[key].forEach((item) => {
+          for (const k in item) {
+            if (typeof item[k] === 'object' && item[k] != null && Object.keys(item[k]).length) {
+              _completeOneLean(item[k], res, opts);
+            }
+            opts.lean.transform({}, item);
```

Comment:
Let's change the `transform()` function so it only takes 1 parameter, `doc`. The first param is always `{}`, which isn't very useful.